developer/frameworks/articles/data-engine/01-data-definition-api/01-data-definition-api-operations.markdown

```diff
@@ -94,10 +105,85 @@ To delete a `DEDataDefinition`, pass a `DEDataDefinitionDeleteRequest` to
 Required parameter: `long dataDefinitionId`
 Optional parameters: none
 
-<!-- API NOT READY FOR DOCS YET -->
-<!--
-## Defining Permissions on Data Definitions
+## Granting and Revoking Permissions on Data Definitions
+
+The permissions operations for `DEDataDefinition`s work similarly to the other
+operations. Pass a `DEDataDefinition*PermissionsRequest` to
+`DEDataDefinitionService.execute`. There are permissions for granting and
+revoking permissions on Data Definitions.
+
+### Granting and Revoking Permission to Add Data Definitions and Manage Permissions
+
+`DEDataDefinitionSavePermissionsRequest` is for granting one or more Roles
+permission to add a new Data Definition to the system, and/or the ability to
+manage User permissions on Data Definitions.
+
+Required parameters: 
+- `long companyId`
+- `long scopeGroupId`
+- `String[] roleNames`
+
+Optional parameters: none
+
+Optional builder methods:
+- `allowAddDefinition()` to grant Add permissions
+- `allowDefinePermissions()` to grant permissions to grant permissions
+<!-- Find a better way to say the above -->
+
+`DEDataDefinitionDeletePermissionsRequest` is for revoking the same permissions
+granted by the `DeDataDefinitionSavePermissionsRequest`. 
+
+Required parameters: 
+- `long companyId`
+- `long scopeGroupId`
+- `String[] roleNames`
+
+Optional parameters: none
+
+Once the `DEDataDefinitionDeletePermissionsRequest` is executed, all specified
+Roles lose both permissions simultaneously: Role Users can no long add Data
+Definitions or manage permissions for Data Definitions. 
+
+### Granting and Revoking Permission to View, Update, or Delete a specific Data Definition
+
+`DEDataDefinitionSaveModelPermissionsRequest` is for granting permission to
+update, view, and delete a specific Data Definition already in the system. 
+
+Required parameters:
+- `long companyId`
+- `long groupId`
+- `long scopeUserId`
+- `long scopeGroupId`
+- `long deDataDefinitionId`
+- `String[] roleNames`
+
+Optional parameters: none
+
+Optional builder methods:
+- `allowDelete()` to grant deletion permission
+- `allowUpdate()` to grant update permission
+- `allowView()` to grant view permission
+
+Specify more than one builder method to grant mutliple permissions
+simultaneously.
+
+Here's an example request builder that grants Site Members all three permissions:
```

Comment:
where?

Reply:
Oops, I moved it to the top of the article :)